examples/simulate_fight_scenario.py

```diff
@@ -0,0 +1,111 @@
+from sc2 import maps
+from sc2.bot_ai import BotAI
+from sc2.data import Race, Difficulty
+from sc2.ids.unit_typeid import UnitTypeId
+from sc2.main import run_game
+from sc2.player import Bot, Computer
+from loguru import logger
+
+
+class FightBot(BotAI):
+    def __init__(self):
+        super().__init__()
+        self.control_requested = False
+        self.control_received = False
+        self.fight_started = False
+        self.supplies_been_damaged = False
+
+    async def on_step(self, iteration):
+        # prepare the level
+        if not self.control_requested:
+            # we need this one for `self.enemy_units` to "see" all units
+            await self._client.debug_show_map()
+            # this one will allow us to do something like: `self.enemy_units.first.attack(self.townhalls.first)`
+            await self._client.debug_control_enemy()
+            logger.info("control requested")
+            # await self.chat_send("control requested")
+            self.control_requested = True
+
+        if self.control_requested and self.enemy_units and not self.control_received:
+            # prepare my side
+            me = 1
+            cc = self.townhalls.first
+            p = cc.position.towards(self.game_info.map_center, 3)
+            # create supply
+            await self._client.debug_create_unit([[UnitTypeId.SUPPLYDEPOT, 1, p, me]])
+            # destroy command center
+            await self._client.debug_kill_unit([cc.tag])
+            # destroy all workers
+            for w in self.units(UnitTypeId.SCV):
+                await self._client.debug_kill_unit([w.tag])
+            # create marines
+            await self._client.debug_create_unit([[UnitTypeId.MARINE, 4, p, me]])
+
+            # prepare opponent side
+            pc = 2
+            cc = self.enemy_structures.first
+            p = cc.position.towards(self.game_info.map_center, 3)
+            # create supply
+            await self._client.debug_create_unit([[UnitTypeId.SUPPLYDEPOT, 1, p, pc]])
+            # destroy command center
+            await self._client.debug_kill_unit([cc.tag])
+            # destroy all workers
+            for w in self.enemy_units(UnitTypeId.SCV):
+                await self._client.debug_kill_unit([w.tag])
+            # create marines
+            await self._client.debug_create_unit([[UnitTypeId.MARINE, 4, p, pc]])
+            logger.info("control received")
+            # await self.chat_send("control received")
+            self.control_received = True
+
+        # to speedup, we are going damage both supplies
+        if not self.supplies_been_damaged and self.structures(UnitTypeId.SUPPLYDEPOT) and self.enemy_structures(UnitTypeId.SUPPLYDEPOT):
+            for s in self.structures:
+                await self._client.debug_set_unit_value([s.tag], 2, 100)
+            for s in self.enemy_structures:
+                await self._client.debug_set_unit_value([s.tag], 2, 100)
```

Comment:
True, in our case if the command center is not destroyed yet, it will be also damaged, but at the very end each player will have only one building - which is supply

Here I see two options: 
- iterate over supply depots only (but still what if there were some depots already)
- remove this optional example at all (we have check to quit game fast so there is no need in this)